telnet/src/lib.rs

```diff
@@ -72,58 +65,40 @@ impl TelnetProvider {
             &config.module,
             self.dispatcher.clone(),
             self.outbounds.clone(),
-        )?;
+        );
 
         Ok(vec![])
     }
 
-    fn deconfigure(&self, _config: CapabilityConfiguration) -> Result<Vec<u8>, Box<dyn Error>> {
+    fn deconfigure(&self, _config: CapabilityConfiguration) -> MessageHandlerResult {
         // Handle removal of resources claimed by an actor here
         // TODO: terminate the telnet server for this actor
         Ok(vec![])
     }
 
     /// Sends a text message to the appropriate socket
-    fn send_text(&self, _actor: &str, msg: TelnetMessage) -> Result<Vec<u8>, Box<dyn Error>> {
+    fn send_text(&self, _actor: &str, msg: TelnetMessage) -> MessageHandlerResult {
         let outbound = self.outbounds.read().unwrap()[&msg.session].clone();
         outbound.send(msg.text).unwrap();
         Ok(vec![])
     }
 
-    fn get_descriptor(&self) -> Result<Vec<u8>, Box<dyn Error>> {
-        Ok(serialize(
-            CapabilityDescriptor::builder()
-                .id(CAPABILITY_ID)
-                .name("waSCC Telnet Server Capability Provider")
-                .long_description(
-                    "A simple telnet server to allow waSCC actors to communicate via telnet",
-                )
-                .version(VERSION)
-                .revision(REVISION)
-                .with_operation(
-                    OP_SESSION_STARTED,
-                    OperationDirection::ToActor,
-                    "Notifies an actor that a new telnet session has started",
-                )
-                .with_operation(
-                    OP_SEND_TEXT,
-                    OperationDirection::ToProvider,
-                    "Sends text from an actor to a telnet session managed by the provider",
-                )
-                .with_operation(
-                    OP_RECEIVE_TEXT,
-                    OperationDirection::ToActor,
-                    "Delivers a single line of text to an actor, tagged with the session ID",
-                )
-                .build(),
-        )?)
+    fn health(&self) -> MessageHandlerResult {
+        let bytes = serialize(HealthCheckResponse {
```

Comment:
This can be converted to just using the `serialize` call without the `Ok` wrapper.

Reply:
fixed in https://github.com/wasmCloud/capability-providers/pull/51/commits/cb89190ff562b02078e9e3424a2bc62db4108fae